src/libsyntax_pos/span_encoding.rs

```diff
@@ -25,10 +25,19 @@ use std::cell::RefCell;
 /// The primary goal of `Span` is to be as small as possible and fit into other structures
 /// (that's why it uses `packed` as well). Decoding speed is the second priority.
 /// See `SpanData` for the info on span fields in decoded representation.
+
+/// Don't use packed attribute on sparc64/v9, see: #45509
+#[cfg(not(any(target_arch = "sparc64",
+              target_arch = "sparcv9")))]
 #[derive(Clone, Copy, PartialEq, Eq, Hash)]
 #[repr(packed)]
 pub struct Span(u32);
 
+#[cfg(any(target_arch = "sparc64",
+          target_arch = "sparcv9"))]
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Span(u32);
```

Comment:
You could simplify this whole thing into:

```rust
#[derive(Clone, Copy, PartialEq, Eq, Hash)]
#[cfg_attr(not(any(target_arch = "sparc64", target_arch = "sparcv9")), repr(packed))]
pub struct Span(u32);
```

Reply:
Thanks! I was expecting that there would be a simpler way. I'll try it out.